### ISSUE TITLE
Support for current Scala.js versions?

### ISSUE THREAD

**Issue by afa-unibas** [antonkulaga/scala-js-binding#22]
Hello, 
I tried including this library in a little project and hit the following error. Is there something I can do about it?

Kind thanks for any answer.

```scala
org.scalajs.core.ir.IRVersionNotSupportedException: Failed to deserialize info of file compiled with Scala.js 0.6.14 (supported: 0.6.0, 0.6.3, 0.6.4, 0.6.6, 0.6.8, 0.6.5): /home/afa/.ivy2/cache/org.denigma/binding_sjs0.6_2.11/jars/binding_sjs0.6_2.11-0.8.17.jar:org/denigma/binding/views/ViewInjector.sjsir
    at org.scalajs.core.tools.io.VirtualSerializedScalaJSIRFile$class.info(VirtualFiles.scala:148)
    at org.scalajs.core.tools.io.MemVirtualSerializedScalaJSIRFile.info(MemFiles.scala:104)
    at org.scalajs.core.tools.io.IRFileCache$PersistentIRFile.<init>(IRFileCache.scala:246)
    at org.scalajs.core.tools.io.IRFileCache$PersistedFiles$$anonfun$update$4.apply(IRFileCache.scala:221)
    at org.scalajs.core.tools.io.IRFileCache$PersistedFiles$$anonfun$update$4.apply(IRFileCache.scala:221)
    at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:244)
    at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:244)
    at scala.collection.immutable.List.foreach(List.scala:318)
    at scala.collection.TraversableLike$class.map(TraversableLike.scala:244)
    at scala.collection.AbstractTraversable.map(Traversable.scala:105)
    at org.scalajs.core.tools.io.IRFileCache$PersistedFiles.update(IRFileCache.scala:221)
    at org.scalajs.core.tools.io.IRFileCache$Cache$$anonfun$update$1$$anonfun$apply$mcV$sp$1.apply(IRFileCache.scala:98)
    at org.scalajs.core.tools.io.IRFileCache$Cache$$anonfun$update$1$$anonfun$apply$mcV$sp$1.apply(IRFileCache.scala:85)
    at scala.collection.immutable.List.foreach(List.scala:318)
    at org.scalajs.core.tools.io.IRFileCache$Cache$$anonfun$update$1.apply$mcV$sp(IRFileCache.scala:85)
    at org.scalajs.core.tools.io.IRFileCache$Cache$$anonfun$update$1.apply(IRFileCache.scala:82)
    at org.scalajs.core.tools.io.IRFileCache$Cache$$anonfun$update$1.apply(IRFileCache.scala:82)
    at org.scalajs.core.tools.io.IRFileCache.org$scalajs$core$tools$io$IRFileCache$$clearOnThrow(IRFileCache.scala:282)
    at org.scalajs.core.tools.io.IRFileCache$Cache.update(IRFileCache.scala:82)
    at org.scalajs.core.tools.io.IRFileCache$Cache.cached(IRFileCache.scala:78)
    at org.scalajs.sbtplugin.ScalaJSPluginInternal$$anonfun$21$$anonfun$apply$17.apply(ScalaJSPluginInternal.scala:387)
    at org.scalajs.sbtplugin.ScalaJSPluginInternal$$anonfun$21$$anonfun$apply$17.apply(ScalaJSPluginInternal.scala:387)
    at sbt.Attributed.map(Attributes.scala:199)
    at org.scalajs.sbtplugin.ScalaJSPluginInternal$$anonfun$21.apply(ScalaJSPluginInternal.scala:387)
    at org.scalajs.sbtplugin.ScalaJSPluginInternal$$anonfun$21.apply(ScalaJSPluginInternal.scala:377)
    at scala.Function1$$anonfun$compose$1.apply(Function1.scala:47)
    at sbt.$tilde$greater$$anonfun$$u2219$1.apply(TypeFunctions.scala:40)
    at sbt.std.Transform$$anon$4.work(System.scala:63)
    at sbt.Execute$$anonfun$submit$1$$anonfun$apply$1.apply(Execute.scala:228)
    at sbt.Execute$$anonfun$submit$1$$anonfun$apply$1.apply(Execute.scala:228)
    at sbt.ErrorHandling$.wideConvert(ErrorHandling.scala:17)
    at sbt.Execute.work(Execute.scala:237)
    at sbt.Execute$$anonfun$submit$1.apply(Execute.scala:228)
    at sbt.Execute$$anonfun$submit$1.apply(Execute.scala:228)
    at sbt.ConcurrentRestrictions$$anon$4$$anonfun$1.apply(ConcurrentRestrictions.scala:159)
    at sbt.CompletionService$$anon$2.call(CompletionService.scala:28)
    at java.util.concurrent.FutureTask.run(FutureTask.java:266)
    at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
    at java.util.concurrent.FutureTask.run(FutureTask.java:266)
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
    at java.lang.Thread.run(Thread.java:748)
Caused by: org.scalajs.core.ir.IRVersionNotSupportedException: This version (0.6.14) of Scala.js IR is not supported. Supported versions are: 0.6.0, 0.6.3, 0.6.4, 0.6.6, 0.6.8, 0.6.5
    at org.scalajs.core.ir.InfoSerializers$Deserializer.readHeader(InfoSerializers.scala:157)
    at org.scalajs.core.ir.InfoSerializers$Deserializer.deserialize(InfoSerializers.scala:100)
    at org.scalajs.core.ir.InfoSerializers$.deserializeWithVersion(InfoSerializers.scala:35)
    at org.scalajs.core.ir.InfoSerializers$.deserialize(InfoSerializers.scala:31)
    at org.scalajs.core.tools.io.VirtualSerializedScalaJSIRFile$class.info(VirtualFiles.scala:145)
    at org.scalajs.core.tools.io.MemVirtualSerializedScalaJSIRFile.info(MemFiles.scala:104)
    at org.scalajs.core.tools.io.IRFileCache$PersistentIRFile.<init>(IRFileCache.scala:246)
    at org.scalajs.core.tools.io.IRFileCache$PersistedFiles$$anonfun$update$4.apply(IRFileCache.scala:221)
    at org.scalajs.core.tools.io.IRFileCache$PersistedFiles$$anonfun$update$4.apply(IRFileCache.scala:221)
    at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:244)
    at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:244)
    at scala.collection.immutable.List.foreach(List.scala:318)
    at scala.collection.TraversableLike$class.map(TraversableLike.scala:244)
    at scala.collection.AbstractTraversable.map(Traversable.scala:105)
    at org.scalajs.core.tools.io.IRFileCache$PersistedFiles.update(IRFileCache.scala:221)
    at org.scalajs.core.tools.io.IRFileCache$Cache$$anonfun$update$1$$anonfun$apply$mcV$sp$1.apply(IRFileCache.scala:98)
    at org.scalajs.core.tools.io.IRFileCache$Cache$$anonfun$update$1$$anonfun$apply$mcV$sp$1.apply(IRFileCache.scala:85)
    at scala.collection.immutable.List.foreach(List.scala:318)
    at org.scalajs.core.tools.io.IRFileCache$Cache$$anonfun$update$1.apply$mcV$sp(IRFileCache.scala:85)
    at org.scalajs.core.tools.io.IRFileCache$Cache$$anonfun$update$1.apply(IRFileCache.scala:82)
    at org.scalajs.core.tools.io.IRFileCache$Cache$$anonfun$update$1.apply(IRFileCache.scala:82)
    at org.scalajs.core.tools.io.IRFileCache.org$scalajs$core$tools$io$IRFileCache$$clearOnThrow(IRFileCache.scala:282)
    at org.scalajs.core.tools.io.IRFileCache$Cache.update(IRFileCache.scala:82)
    at org.scalajs.core.tools.io.IRFileCache$Cache.cached(IRFileCache.scala:78)
    at org.scalajs.sbtplugin.ScalaJSPluginInternal$$anonfun$21$$anonfun$apply$17.apply(ScalaJSPluginInternal.scala:387)
    at org.scalajs.sbtplugin.ScalaJSPluginInternal$$anonfun$21$$anonfun$apply$17.apply(ScalaJSPluginInternal.scala:387)
    at sbt.Attributed.map(Attributes.scala:199)
    at org.scalajs.sbtplugin.ScalaJSPluginInternal$$anonfun$21.apply(ScalaJSPluginInternal.scala:387)
    at org.scalajs.sbtplugin.ScalaJSPluginInternal$$anonfun$21.apply(ScalaJSPluginInternal.scala:377)
    at scala.Function1$$anonfun$compose$1.apply(Function1.scala:47)
    at sbt.$tilde$greater$$anonfun$$u2219$1.apply(TypeFunctions.scala:40)
    at sbt.std.Transform$$anon$4.work(System.scala:63)
    at sbt.Execute$$anonfun$submit$1$$anonfun$apply$1.apply(Execute.scala:228)
    at sbt.Execute$$anonfun$submit$1$$anonfun$apply$1.apply(Execute.scala:228)
    at sbt.ErrorHandling$.wideConvert(ErrorHandling.scala:17)
    at sbt.Execute.work(Execute.scala:237)
    at sbt.Execute$$anonfun$submit$1.apply(Execute.scala:228)
    at sbt.Execute$$anonfun$submit$1.apply(Execute.scala:228)
    at sbt.ConcurrentRestrictions$$anon$4$$anonfun$1.apply(ConcurrentRestrictions.scala:159)
    at sbt.CompletionService$$anon$2.call(CompletionService.scala:28)
    at java.util.concurrent.FutureTask.run(FutureTask.java:266)
    at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
    at java.util.concurrent.FutureTask.run(FutureTask.java:266)
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
    at java.lang.Thread.run(Thread.java:748)
[error] (frontend/compile:scalaJSIR) org.scalajs.core.ir.IRVersionNotSupportedException: Failed to deserialize info of file compiled with Scala.js 0.6.14 (supported: 0.6.0, 0.6.3, 0.6.4, 0.6.6, 0.6.8, 0.6.5): /home/afa/.ivy2/cache/org.denigma/binding_sjs0.6_2.11/jars/binding_sjs0.6_2.11-0.8.17.jar:org/denigma/binding/views/ViewInjector.sjsir
```

**Comment by antonkulaga** [comment]
I have to see your sbt file to make any comments

**Comment by ioolkos** [comment]
It is a template from a Udash framework project: (see the following 2 files).
Please don't exclude I might make a stupid error, as I just started out with this.

Thank you very much for your time and comments! 

```scala
import com.lihaoyi.workbench.Plugin._
import UdashBuild._
import Dependencies._

name := "neye"

version in ThisBuild := "0.1.0-SNAPSHOT"
scalaVersion in ThisBuild := "2.11.8"
organization in ThisBuild := ""
crossPaths in ThisBuild := false
scalacOptions in ThisBuild ++= Seq(
  "-feature",
  "-deprecation",
  "-unchecked",
  "-language:implicitConversions",
  "-language:existentials",
  "-language:dynamics",
  "-Xfuture",
  "-Xfatal-warnings",
  "-Xlint:_,-missing-interpolator,-adapted-args"
)

def crossLibs(configuration: Configuration) =
  libraryDependencies ++= crossDeps.value.map(_ % configuration)

lazy val neye = project.in(file("."))
  .aggregate(sharedJS, sharedJVM, frontend, backend)
  .dependsOn(backend)
  .settings(
    publishArtifact := false,
    mainClass in Compile := Some("neye.Launcher")
  )

lazy val shared = crossProject.crossType(CrossType.Pure).in(file("shared"))
  .settings(
    crossLibs(Provided)
  )

lazy val sharedJVM = shared.jvm
lazy val sharedJS = shared.js

lazy val backend = project.in(file("backend"))
  .dependsOn(sharedJVM)
  .settings(
    libraryDependencies ++= backendDeps.value,
    crossLibs(Compile),

    compile <<= (compile in Compile),
    (compile in Compile) <<= (compile in Compile).dependsOn(copyStatics),
    copyStatics := IO.copyDirectory((crossTarget in frontend).value / StaticFilesDir, (target in Compile).value / StaticFilesDir),
    copyStatics <<= copyStatics.dependsOn(compileStatics in frontend),

    mappings in (Compile, packageBin) ++= {
      copyStatics.value
      ((target in Compile).value / StaticFilesDir).***.get map { file =>
        file -> file.getAbsolutePath.stripPrefix((target in Compile).value.getAbsolutePath)
      }
    },

    watchSources ++= (sourceDirectory in frontend).value.***.get
  )

lazy val frontend = project.in(file("frontend")).enablePlugins(ScalaJSPlugin)
  .dependsOn(sharedJS)
  .settings(
      resolvers += sbt.Resolver.bintrayRepo("denigma", "denigma-releases"),
      libraryDependencies ++= frontendDeps.value,
    crossLibs(Compile),
    jsDependencies ++= frontendJSDeps.value,
    persistLauncher in Compile := true,

    compile <<= (compile in Compile).dependsOn(compileStatics),
    compileStatics := {
      IO.copyDirectory(sourceDirectory.value / "main/assets/fonts", crossTarget.value / StaticFilesDir / WebContent / "assets/fonts")
      IO.copyDirectory(sourceDirectory.value / "main/assets/images", crossTarget.value / StaticFilesDir / WebContent / "assets/images")
      val statics = compileStaticsForRelease.value
      (crossTarget.value / StaticFilesDir).***.get
    },

    artifactPath in(Compile, fastOptJS) :=
      (crossTarget in(Compile, fastOptJS)).value / StaticFilesDir / WebContent / "scripts" / "frontend-impl-fast.js",
    artifactPath in(Compile, fullOptJS) :=
      (crossTarget in(Compile, fullOptJS)).value / StaticFilesDir / WebContent / "scripts" / "frontend-impl.js",
    artifactPath in(Compile, packageJSDependencies) :=
      (crossTarget in(Compile, packageJSDependencies)).value / StaticFilesDir / WebContent / "scripts" / "frontend-deps-fast.js",
    artifactPath in(Compile, packageMinifiedJSDependencies) :=
      (crossTarget in(Compile, packageMinifiedJSDependencies)).value / StaticFilesDir / WebContent / "scripts" / "frontend-deps.js",
    artifactPath in(Compile, packageScalaJSLauncher) :=
      (crossTarget in(Compile, packageScalaJSLauncher)).value / StaticFilesDir / WebContent / "scripts" / "frontend-init.js"
  ).settings(workbenchSettings:_*)
  .settings(
    bootSnippet := "neye.Init().main();",
    updatedJS := {
      var files: List[String] = Nil
      ((crossTarget in Compile).value / StaticFilesDir ** "*.js").get.foreach {
        (x: File) =>
          streams.value.log.info("workbench: Checking " + x.getName)
          FileFunction.cached(streams.value.cacheDirectory / x.getName, FilesInfo.lastModified, FilesInfo.lastModified) {
            (f: Set[File]) =>
              val fsPath = f.head.getAbsolutePath.drop(new File("").getAbsolutePath.length)
              files = "http://localhost:12345" + fsPath :: files
              f
          }(Set(x))
      }
      files
    },
    //// use either refreshBrowsers OR updateBrowsers
    // refreshBrowsers <<= refreshBrowsers triggeredBy (compileStatics in Compile)
    updateBrowsers <<= updateBrowsers triggeredBy (compileStatics in Compile)
  )
```
There is also this "Dependencies object": 

```scala
import org.scalajs.sbtplugin.ScalaJSPlugin.autoImport._
import sbt._

object Dependencies {
    val udashVersion = "0.4.0"
    val udashJQueryVersion = "1.0.0"
    val logbackVersion = "1.1.3"
    val jettyVersion = "9.3.11.v20160721"

    val crossDeps = Def.setting(Seq[ModuleID](
        "io.udash" %%% "udash-core-shared" % udashVersion,
        "io.udash" %%% "udash-rpc-shared" % udashVersion
    ))

    val frontendDeps = Def.setting(Seq[ModuleID](
        "io.udash" %%% "udash-core-frontend" % udashVersion,
        "io.udash" %%% "udash-jquery" % udashJQueryVersion,
        "io.udash" %%% "udash-rpc-frontend" % udashVersion,
        "com.github.japgolly.scalacss" %%% "core" % "0.5.0",
        "com.github.japgolly.scalacss" %%% "ext-scalatags" % "0.5.0",
        "io.udash" %%% "udash-bootstrap" % "0.4.0",
        "org.denigma" %%% "codemirror-facade" % "5.11-0.7",
        "org.denigma" %%% "binding" % "0.8.17",
        "org.denigma" %%% "binding-controls" % "0.0.25"
    ))

    val frontendJSDeps = Def.setting(Seq[org.scalajs.sbtplugin.JSModuleID](
    ))

    val backendDeps = Def.setting(Seq[ModuleID](
        "ch.qos.logback" % "logback-classic" % logbackVersion,
        "org.eclipse.jetty" % "jetty-server" % jettyVersion,
        "org.eclipse.jetty" % "jetty-servlet" % jettyVersion,
        "io.udash" %% "udash-rpc-backend" % udashVersion,
        "org.eclipse.jetty.websocket" % "websocket-server" % jettyVersion
    ))
}
```

**Comment by afa-unibas** [comment]
Seems I resolved this somehow by going back to Scala 2.11.8. Thanks!
